enforcer.go

```diff
@@ -459,9 +477,30 @@ func (e *Enforcer) enforce(matcher string, explains *[]string, rvals ...interfac
 			functions[key] = util.GenerateGFunction(rm)
 		}
 	}
+
+	var (
+		rType = "r"
```

Comment:
This variable declaration will allocate heap memory everytime we call `enforce ` method. I think It's better if we can define these values as constants so it will be created in the stack.